Source/Scenes/Overworld.cs

```diff
@@ -199,8 +215,59 @@ public Overworld(bool startOnLastSelected)
 		}
 
 		cameraCloseUpEase = 1.0f;
+
+		entries = GetCurrentModEntries();
+	}
+	#endregion
+
+	#region Overworld Methods
+	public List<Entry> GetCurrentModEntries()
+	{
+		List<Entry> entriesTemp = [];
+
+		// We treat the vanilla "mod" as a catch-all option (because it only has one level anyway). 
+		// It will return every available item.
+		if (selectedMod is VanillaGameMod)
+		{
+			foreach (var level in Assets.Levels)
+			{
+				var mod = ModManager.Instance.Mods.FirstOrDefault(mod => mod.Levels.Contains(level));
+				if (mod is { Enabled: true })
+				{
+					entriesTemp.Add(new(level, mod));
+				}
+			}
+		}
+		else
+		{
+			foreach (var level in selectedMod.Levels)
+			{
+				entriesTemp.Add(new(level, selectedMod));
+			}
+		}
+
+		return entriesTemp;
+	}
+
+	// reference taken from Menu.OptionList
```

Comment:
minor nitpick. I feel like the implementation here ended up pretty different from the OptionList, so this comment probably isn't really necessary.